src/main/java/org/spongepowered/common/statistic/SpongeBlockStatistic.java

```diff
@@ -41,14 +43,22 @@
 public final class SpongeBlockStatistic extends StatCrafting implements BlockStatistic, TypedSpongeStatistic {
 
     private String spongeId;
+    private final Translation translation;
 
     public SpongeBlockStatistic(String statId, String itemName, ITextComponent statName, Item item) {
         super(statId, itemName, statName, item);
+        String args;
+        if (statId.equals("stat.mineBlock.")) {
+            args = Block.getBlockFromItem(item).getLocalizedName();
+        } else {
+            args = new ItemStack(item).getDisplayName();
+        }
+        this.translation = new SpongeTranslation(statId.substring(0, statId.length() - 1), args);
     }
 
     @Override
     public Translation getTranslation() {
-        return new SpongeTranslation(this.statId);
+        return this.translation;
```

Comment:
Does this have to be field initialized in the constructor? You're doing lazy loading in pretty much other places, might as well lazy load from here, no?

Reply:
I'm not sure there's a clean way to do this. I need statId and item.